axiom/client.py

```diff
@@ -71,11 +167,162 @@ def __init__(self, token: str, org_id: str = None, url_base: str = AXIOM_URL):
+            self.session.headers.update({"X-Axiom-Org-Id": org_id})
+
+        self.datasets = DatasetsClient(self.session, self.logger)
+        self.users = UsersClient(self.session)
+
+    def ingest(
+        self,
+        dataset: str,
+        payload: bytes,
+        contentType: ContentType,
+        enc: ContentEncoding,
+        opts: Optional[IngestOptions] = None,
+    ) -> IngestStatus:
+        """Ingest the events into the named dataset and returns the status."""
+        path = "datasets/%s/ingest" % dataset
+
+        # check if passed content type and encoding are correct
+        if not contentType:
+            raise ValueError("unknown content-type, choose one of json,x-ndjson or csv")
+
+        if not enc:
+            raise ValueError("unknown content-encoding")
+
+        # set headers
+        headers = {"Content-Type": contentType.value, "Content-Encoding": enc.value}
+        # prepare query params
+        params = self._prepare_ingest_options(opts)
+
+        # override the default header and set the value from the passed parameter
+        res = self.session.post(path, data=payload, headers=headers, params=params)
+        status_snake = decamelize(res.json())
+        return Util.from_dict(IngestStatus, status_snake)
+
+    def ingest_events(
+        self,
+        dataset: str,
+        events: List[dict],
+        opts: Optional[IngestOptions] = None,
+    ) -> IngestStatus:
+        """Ingest the events into the named dataset and returns the status."""
+        # encode request payload to NDJSON
+        content = ndjson.dumps(events).encode("UTF-8")
+        gzipped = gzip.compress(content)
+
+        return self.ingest(
+            dataset, gzipped, ContentType.NDJSON, ContentEncoding.GZIP, opts
+        )
+
+    def query_legacy(
+        self, id: str, query: QueryLegacy, opts: QueryOptions
+    ) -> QueryLegacyResult:
+        """Executes the given query on the dataset identified by its id."""
+        if not opts.saveAsKind or (opts.saveAsKind == QueryKind.APL):
+            raise WrongQueryKindException(
+                "invalid query kind %s: must be %s or %s"
+                % (opts.saveAsKind, QueryKind.ANALYTICS, QueryKind.STREAM)
+            )
+
+        path = "datasets/%s/query" % id
+        payload = ujson.dumps(asdict(query), default=Util.handle_json_serialization)
+        self.logger.debug("sending query %s" % payload)
+        params = self._prepare_query_options(opts)
+        res = self.session.post(path, data=payload, params=params)
+        result = Util.from_dict(QueryLegacyResult, res.json())
+        self.logger.debug(f"query result: {result}")
+        query_id = res.headers.get("X-Axiom-History-Query-Id")
+        self.logger.info(f"received query result with query_id: {query_id}")
+        result.savedQueryID = query_id
+        return result
+
+    def apl_query(self, apl: str, opts: AplOptions) -> QueryResult:
```

Comment:
```suggestion
    def query(self, apl: str, opts: AplOptions = {}) -> QueryResult:
```

Reply:
there is already a function called query, we kept `apl_query` to maintain backwards compatibility, the `apl_query()` points to `query()`